library/psa_crypto.c

```diff
@@ -2607,6 +2624,17 @@ psa_status_t psa_hash_setup( psa_hash_operation_t *operation,
         return( PSA_ERROR_BAD_STATE );
     }
 
+    operation->mbedtls_in_use = 0;
```

Comment:
This is not aligned with the organization of the PSA implementation with respect to the PSA driver interface that we are currently trying to set-up. The code organization is described in the psa-crypto-implementation-structure.md document in PR #3933 and an example of its application for multi-part operation can be found [here](https://github.com/ronald-cron-arm/mbedtls/tree/psa-cipher) (for ciphers).
Thus this is likely that this PR will need some significant refactoring. It is also probably better to not start any refactoring before PR #3933 is merged.